lib/ibandit/iban.rb

```diff
@@ -276,19 +276,47 @@ def supports_iban_determination?
     def valid_swedish_details?
       return true unless country_code == 'SE'
 
-      bank_details = {
-        bank_code: swift_bank_code,
-        account_number: swift_account_number
-      }
+      if branch_code
```

Comment:
Yeah, validation in this class feels like a larger problem. Didn't want to introduce inconsistency here so used `SwedishDetailsValidator` just for validation helper methods. Happy to revisit, but probably best to do in another PR?